core/src/wallet/cosmos/CosmosLikeAccount.cpp

```diff
@@ -395,28 +393,31 @@ int CosmosLikeAccount::putTransaction(soci::session &sql, const cosmos::Transact
             continue;
         }
 
-        CosmosLikeOperation operation(tx, msg);
+        CosmosLikeOperation operation(tx, msg, log);
         inflateOperation(operation, getWallet(), tx, msg);
         operation.refreshUid(std::to_string(msgIndex));
-
-        auto inserted = CosmosLikeOperationDatabaseHelper::putOperation(sql, operation);
-        if (inserted) {
-            CosmosLikeOperationDatabaseHelper::updateOperation(sql, operation.uid, msg.uid);
-            emitNewOperationEvent(operation);
-        }
-        result = static_cast<int>(operation.type);
+        out.push_back(operation);
     }
+}
 
-    return result;
+Try<int> CosmosLikeAccount::bulkInsert(const std::vector<CosmosLikeOperation> &operations) {
+    return Try<int>::from([&] () {
+        soci::session sql(getWallet()->getDatabase()->getPool());
+        soci::transaction tr(sql);
+        CosmosLikeOperationsDatabaseHelper::bulkInsert(sql, operations);
+        tr.commit();
+        // Emit
+        for (const auto& op : operations) {
+            emitNewOperationEvent(op);
+        }
+        return operations.size();
+    });
 }
 
 bool CosmosLikeAccount::putBlock(soci::session &sql, const api::Block &block)
 {
-    if (BlockDatabaseHelper::putBlock(sql, block)) {
-        emitNewBlockEvent(block);
-        return true;
-    }
-    return false;
+    emitNewBlockEvent(block);
```

Comment:
I've discussed with pierre. 
we should emit NewBlockEvent only if the block doesn't exist in DB. So I think we should keep the old logic.